File: WorldPainter/WPCore/src/main/java/org/pepsoft/worldpainter/exporting/gpuacceleration/GPUMemoryBlock.java
package org.pepsoft.worldpainter.exporting.gpuacceleration;

public class GPUMemoryBlock{
    //array of size of number of gpu memory allocations that contains an array of pointers to GPU memory.
    private final long xGPUPointer;
    private final long yGPUPointer;
    private final long zGPUPointer;
    private final long pGPUPointer;
    private final long outputGPUPointer;
    private final long compactedOutputGPUPointer;

    public GPUMemoryBlock(long xGPUPointer, long yGPUPointer, long zGPUPointer, long pGPUPointer, long outputGPUPointer, long compactedOutputGPUPointer) {
        this.xGPUPointer = xGPUPointer;
        this.yGPUPointer = yGPUPointer;
        this.zGPUPointer = zGPUPointer;
        this.pGPUPointer = pGPUPointer;
        this.outputGPUPointer = outputGPUPointer;
        this.compactedOutputGPUPointer = compactedOutputGPUPointer;
    }

    public GPUMemoryBlock() {
        this(0L,0L,0L,0L,0L,0L);
    }

    public long getxGPUPointer() {
        return xGPUPointer;
    }

    public long getyGPUPointer() {
        return yGPUPointer;
    }

    public long getzGPUPointer() {
        return zGPUPointer;
    }

    public long getpGPUPointer() {
        return pGPUPointer;
    }

    public long getOutputGPUPointer() {
        return outputGPUPointer;
    }

    public long getCompactedOutputGPUPointer() {
        return compactedOutputGPUPointer;
    }
}

File: WorldPainter/NoiseHardwareAccelerator/NoiseHardwareAccelerator/JavaClassWrappers/GPUMemoryBlock.cpp
#include "GPUMemoryBlock.h"

GPUMemoryBlock::GPUMemoryBlock(JNIEnv* env) {
    this->env = env;

	gpuMemoryBlockClass = env ->FindClass("org/pepsoft/worldpainter/exporting/gpuacceleration/GPUMemoryBlock");

    getxGPUPointerMethod = env->GetMethodID(gpuMemoryBlockClass, "getxGPUPointer", "()J");
    getyGPUPointerMethod = env->GetMethodID(gpuMemoryBlockClass, "getyGPUPointer", "()J");
    getzGPUPointerMethod = env->GetMethodID(gpuMemoryBlockClass, "getzGPUPointer", "()J");
    getpGPUPointerMethod = env->GetMethodID(gpuMemoryBlockClass, "getpGPUPointer", "()J");
    getOutputGPUPointerMethod = env->GetMethodID(gpuMemoryBlockClass, "getOutputGPUPointer", "()J");
    getCompactedOutputGPUPointerMethod = env->GetMethodID(gpuMemoryBlockClass, "getCompactedOutputGPUPointer", "()J");
}

long GPUMemoryBlock::getxGPUPointer() {
    return env->CallLongMethod(gpuMemoryBlockClass, getxGPUPointerMethod);
}

long GPUMemoryBlock::getyGPUPointer() {
    return env->CallLongMethod(gpuMemoryBlockClass, getyGPUPointerMethod);
}

long GPUMemoryBlock::getzGPUPointer() {
    return env->CallLongMethod(gpuMemoryBlockClass, getzGPUPointerMethod);
}

long GPUMemoryBlock::getpGPUPointer() {
    return env->CallLongMethod(gpuMemoryBlockClass, getpGPUPointerMethod);
}

long GPUMemoryBlock::getOutputGPUPointer() {
    return env->CallLongMethod(gpuMemoryBlockClass, getOutputGPUPointerMethod);
}

long GPUMemoryBlock::getcompactedOutputGPUPointer() {
    return env->CallLongMethod(gpuMemoryBlockClass, getCompactedOutputGPUPointerMethod);
}